echo_server.py

```diff
@@ -0,0 +1,89 @@
+import socket
+
+
+def create_server_socket():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    # Set up the server socket
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+    return server_socket
+
+
+def response_ok():
+    return """\
+HTTP/1.1 200 OK\r\n\
+Content-Type: text/plain\r\n\r\n\
+Hey you.\r\n\
+"""
```

Comment:
This will not be a properly formatted HTTP response.  Using the python multi-line string ensures that `\n` is already at the end of each line.  So you *could* just use the `\r`, but even that is prone to errors.  Instead, assemble each line separately and append them into a list, then join the list with `"\r\n"`.